src/components/AttachmentList/AttachmentList.jsx

```diff
@@ -1,148 +1,124 @@
-import React from 'react'
+import React, { useContext, useRef, useEffect } from 'react'
 import PropTypes from 'prop-types'
+
 import getValidProps from '@helpscout/react-utils/dist/getValidProps'
+
 import Attachment from '../Attachment'
 import Icon from '../Icon'
-import Inline from '../Inline'
 import Overflow from '../Overflow'
 import classNames from 'classnames'
-import { AttachmentListUI } from './AttachmentList.css'
+import { AttachmentListUI, ContentUI } from './AttachmentList.css'
+import { AttachmentContext } from '../Attachment/Attachment.Provider'
+import IconButton from '../IconButton'
 
 function noop() {}
 
-export class AttachmentList extends React.Component {
-  static contextTypes = {
-    theme: () => null,
-  }
-
-  handleApplyFade = () => {}
-  handleScrollToEnd = () => {}
-
-  componentDidMount() {
-    this.handleApplyFade()
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.didAddNewAttachment(prevProps)) {
-      this.handleOnAddNewAttachment()
+export const AttachmentList = props => {
+  const {
+    children,
+    className,
+    downloadAllLabel,
+    onDownloadAllClick,
+    showDownloadAll,
+    withOverflow,
+    onScrollEnd,
+    theme: themeProp,
+    ...rest
+  } = props
+
+  const attachmentChildren = React.Children.toArray(children).filter(
+    child => child.type && child.type === Attachment
+  )
```

Comment:
not really needed, just in case someone do insert something not supported within a list
```javascript
<AttachmentList>
  <Attachment />
  <Button />
  <Attachment />
</AttachmentList>
```